configs/example/gem5_library/fdp-hello.py

```diff
@@ -0,0 +1,333 @@
+            self.l1dcaches[i].mem_side = self.l2buses[i].cpu_side_ports
+
+            self.mmucaches[i].mem_side = self.l2buses[i].cpu_side_ports
+
+            self.mmubuses[i].mem_side_ports = self.mmucaches[i].cpu_side
+            self.l2buses[i].mem_side_ports = self.l2caches[i].cpu_side
+
+            self.membus.cpu_side_ports = self.l2caches[i].mem_side
+
+            cpu.connect_walker_ports(
+                self.mmubuses[i].cpu_side_ports,
+                self.mmubuses[i].cpu_side_ports,
+            )
+
+            if board.get_processor().get_isa() == ISA.X86:
+                int_req_port = self.membus.mem_side_ports
+                int_resp_port = self.membus.cpu_side_ports
+                cpu.connect_interrupt(int_req_port, int_resp_port)
+            else:
+                cpu.connect_interrupt()
+
+
+cache_hierarchy = CacheHierarchy(
+    l1i_size="32KiB", l1d_size="32KiB", l2_size="1MB"
+)
+
+
+# 1. Decoupled front-end ------------------------------------------------
+# First setup the decoupled front-end. Its implemented in the O3 core.
+# Create the processor with one core
+processor = SimpleProcessor(
+    cpu_type=CPUTypes.O3, isa=isa_choices[args.isa], num_cores=1
+)
+cpu = processor.cores[0].core
+
+
+# We need to configure the decoupled front-end with some specific parameters.
+# First the fetch buffer and fetch target size. We want double the size of
+# the fetch buffer to be able to run ahead of fetch
+cpu.fetchBufferSize = 16
+cpu.fetchTargetWidth = 32
+
+# The decoupled front-end leverages the BTB to find branches in the fetch
+# stream. Starting from the end of the last fetch target it will search
+# all addresses until a hit. However, for fixed size instruction architectures
+# like ARM only every n-th address must be checked.
+if args.isa == "Arm":
+    cpu.minInstSize = (
+        4  # Note Arm has a 2 byte thumb mode but we ignore it here
+    )
+elif args.isa == "RiscV":
+    cpu.minInstSize = 4  # RiscV has a 2 byte compressed mode.
+else:  # Variable length ISA (x86) must search every byte
+    cpu.minInstSize = 1
+
+# The `decoupledFrontEnd` parameter enables the decoupled front-end.
+# Disable it to get the baseline.
+if args.disable_fdp:
+    cpu.decoupledFrontEnd = False
+else:
+    cpu.decoupledFrontEnd = True
+
+
+## Branch Predictor Unit --------------------------------
+# The decoupled front-end relies on the BTB to discover
+# branches and uses `takenOnlyHistory`.
+# Also at the moment only LTAGE and TAGE-SC-L are supported.
+class BTB(SimpleBTB):
+    numEntries = 16 * 1024
+    associativity = 8
+
+
+class BPU(LTAGE):
+    instShiftAmt = 2 if args.isa == "Arm" else 0
+    btb = BTB()
+    requiresBTBHit = True
+    takenOnlyHistory = True
+
+
+# Configure the branch predictor
+cpu.branchPred = BPU()
+
+
+print(
+    "Running {} on {}, FDP {}".format(
+        args.workload, args.isa, "disabled" if args.disable_fdp else "enabled"
+    )
```

Comment:
Nit: I prefer f-strings.

```suggestion
    f"Running {args.workload} on {args.isa}, FDP {'disabled' if args.disable_fdp else 'enabled'}".
```